### PR TITLE
Better logging

### PR DESCRIPTION
**Pull request by jarongrigat**
Logging is quite inconsistent and I personally never find what I want at the place I expect it. Some rules that make sense to me:

- Avoid repeating DEBUG-level messages. They are great during development but at the current state they clog the debug logs and are not used anymore.
- Use WARNING for things that are expected but should be in the website logs. Like 'alarm for sensor XXX', 'sending SMS to ...'
- Use WARNING also for things that are somewhat unexpected but don't hinder Doberman's operation too much. E.g. 'didn't get a value from device XXX' which could be a one time thing or more regular -> Good to see these on the website
- Use ERROR for (most) caught exceptions. Here something just did not work as it should. A device/pipeline didn't get started, an alarm message could not be sent, ...

Open for debate:
- Use CRITICAL never?
- difference INFO and DEBUG? Maybe it also makes sense to keep the `self.logger.debug` but not log them unless Doberman is explicitly started in DEBUG mode...

### CODE REVIEW

**Comment by jarongrigat** [comment]
Update on the idea:
DEBUG: periodic/repeating & expected messages (e.g. 'got new value from senor'). Only logged when doberman started with --debug flag)
INFO: single events, expected (e.g. construction pipeline...)
WARNING: works as excepted from Doberman side, but is a problem on experiment side (e.g. alarm for sensor X)
ERROR: problem on Doberman side, but could be a one-time hickup (e.g. 'command not understood', 'failed to ping XY')
CRITICAL: error that prevents a system component from running properly. I.e. an error in the (newly added) code that requires a restart of the system, fix of the code, etc. (e.g. 'thread died', 'something went wrong during initialization of device')

File: Doberman/AlarmMonitor.py [hunk 1]
```diff
@@ -113,25 +99,23 @@ def send_email(self, toaddr, subject, message, level, pipeline, cc=None, bcc=Non
...
                                f'{text}</a></li> '
             message += '</ul>'
-        if add_signature:
-            message += f'<hr>Message created on {now} by Doberman slow control.'
-        body = str(message)
-        msg.attach(MIMEText(body, 'html'))
+        message += f'<hr>Message created on {now} by Doberman slow control.'
+        msg.attach(MIMEText(message, 'html'))
         # Connect and send
+        self.logger.warning(f'Sending e-mail to {len(addresses)} recipient{"s" if len(addresses)>1 else ""}')
```
Comment (molztim) [inline Doberman/AlarmMonitor.py:109]: Why is this a warning? In the context its worth a warning, but wouldn't it be more an Info, if we reserve warnings for hickups?

File: Doberman/DeviceMonitor.py [hunk 2]
```diff
@@ -33,7 +32,7 @@ def start_sensor(self, sensor_name):
                 # the "secondary" multi-sensors store the name of the base
                 sensor = Doberman.MultiSensor(**kwargs)
             else:
-                self.logger.debug(f'Not constructing {sensor_name} because it isn\'t the multi primary')
+                self.logger.info(f'Not constructing {sensor_name} because it isn\'t the multi primary')
```
[the remark this answers is not in the document]
Comment (jarongrigat) [reply Doberman/DeviceMonitor.py:35]: You only want the main sensor of a multi sensor to be constructed. all others should be skipped. this is the expected behavior

File: Doberman/AlarmMonitor.py (hunk 1 above)
Comment (jarongrigat) [reply Doberman/AlarmMonitor.py:109]: Just so it gets logged to the database/ is visible on the website when there is an alarm sent to people.

**Pull request merged**